src/collapse/dtrace.rs

```diff
@@ -0,0 +1,163 @@
+use super::Collapse;
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::collections::{HashMap, VecDeque};
+use std::io;
+use std::io::prelude::*;
+
+const HEADER_LINES: u8 = 3;
+
+lazy_static! {
+    static ref CPP_REGEX: Regex = Regex::new("(?P<n>::.*)[(<].*").unwrap();
+}
+
+/// Settings that change how frames are named from the incoming stack traces.
+///
+/// All options default to off.
+#[derive(Clone, Debug, Default)]
+pub struct Options {}
+
+/// A stack collapser for the output of dtrace `ustace()`.
+///
+/// To construct one, either use `dtrace::Folder::default()` or create an [`Options`] and use
+/// `dtrace::Folder::from(options)`.
+#[derive(Default)]
+pub struct Folder {
+    /// All lines until the next empty line are stack lines.
+    in_event: bool,
```

Comment:
Do we even need this any more?

Reply:
It's part of the after_event loop to prevent printing empty lines in case of a mal-formated (i.e. empty) file.

Reply:
I _think_ with the proposed change where we just look for counts, we can simply ignore empty lines, no?

Reply:
Got what you meant with the loop! removed :)